test/e2e/network/service.go

```diff
@@ -3431,3 +3431,66 @@ func validateEndpointsPorts(c clientset.Interface, namespace, serviceName string
 	}
 	return fmt.Errorf("Timed out waiting for service %s in namespace %s to expose endpoints %v (%v elapsed)", serviceName, namespace, expectedEndpoints, framework.ServiceStartTimeout)
 }
+
+var _ = SIGDescribe("SCTP [Feature:SCTP] [LinuxOnly]", func() {
+	f := framework.NewDefaultFramework("sctp")
+
+	var cs clientset.Interface
+
+	ginkgo.BeforeEach(func() {
+		// requires cloud load-balancer support - this feature currently supported only on GCE/GKE
+		e2eskipper.SkipIfNodeOSDistroIs("windows")
+		if proxyMode, err := proxyMode(f); err == nil {
+			if proxyMode == "userspace" {
+				e2eskipper.Skipf("The test doesn't work with kube-proxy in userspace mode")
+			}
+		} else {
+			framework.Logf("Couldn't detect KubeProxy mode - test failure may be expected: %v", err)
+		}
+
+		cs = f.ClientSet
+	})
+
+	ginkgo.AfterEach(func() {
+		if ginkgo.CurrentGinkgoTestDescription().Failed {
+			DescribeSvc(f.Namespace.Name)
+		}
+	})
+
+	ginkgo.It("should serve a basic SCTP service with pod and endpoints", func() {
+		serviceName := "sctp-endpoint-test"
+		ns := f.Namespace.Name
+		jig := e2eservice.NewTestJig(cs, ns, serviceName)
+
+		ginkgo.By("creating service " + serviceName + " in namespace " + ns)
+		defer func() {
```

Comment:
should this defer go after the L3470 where the service is created?

Reply:
Fixed.

Reply:
my bad, I think that I gave you a wrong review, it should be 

```
_, err = jig.CreateSCTPServiceWithPort(nil, 5060)
framework.ExpectNoError(err)
defer func() {
```
if the service creation errors we shouldn't have to delete it.

Reply:
Fixed.